db/active_replicator_push.go

```diff
@@ -238,26 +209,35 @@ func (apr *ActivePushReplicator) reset() error {
 	if apr.state != ReplicationStateStopped {
 		return fmt.Errorf("reset invoked for replication %s when the replication was not stopped", apr.config.ID)
 	}
-	// TODO: this needs pointing at all collections the replicator is configured for!
-	collection := apr.config.ActiveDB.GetSingleDatabaseCollection()
-	if err := resetLocalCheckpoint(collection.dataStore, apr.CheckpointID); err != nil {
+
+	err := apr.forEachCollection(func(c *activeReplicatorCollection) error {
+		return resetLocalCheckpoint(c.dataStore, apr.CheckpointID)
+	})
+	if err != nil {
 		return err
 	}
 
 	apr.lock.Lock()
-	apr.Checkpointer = nil
+	_ = apr.forEachCollection(func(c *activeReplicatorCollection) error {
+		c.Checkpointer = nil
+		return nil
+	})
 	apr.lock.Unlock()
 
 	return nil
 }
 
 // registerCheckpointerCallbacks registers appropriate callback functions for checkpointing.
-func (apr *ActivePushReplicator) registerCheckpointerCallbacks() {
-	apr.blipSyncContext.sgr2PushAlreadyKnownSeqsCallback = apr.Checkpointer.AddAlreadyKnownSeq
-
-	apr.blipSyncContext.sgr2PushAddExpectedSeqsCallback = apr.Checkpointer.AddExpectedSeqs
+func (apr *ActivePushReplicator) registerCheckpointerCallbacks(c *activeReplicatorCollection) {
+	blipSyncContextCollection, err := apr.blipSyncContext.collections.get(c.collectionIdx)
+	if err != nil {
+		base.WarnfCtx(apr.ctx, "Unable to get blipSyncContextCollection for collection %v", c.collectionIdx)
```

Comment:
This panics in the pull replicator, and maybe it's worth having a fatal error for this in both push and pull?